`make offline` fails on macOS

<!--
Hello and thanks for contributing to the Mapbox Maps SDKs for iOS and macOS! To help us diagnose your problem quickly, please:

 - Include a minimal demonstration of the bug, including code, logs, and screenshots.
 - Ensure you can reproduce the bug using the latest release.
 - Only post to report a bug or request a feature; direct all other questions to: https://stackoverflow.com/questions/tagged/mapbox

Start with a brief description below this line. -->

I get a linker error for `mbgl::LayerManager::get()` when trying to build the `mbgl-offline` tool.

### Steps to reproduce

 1. Clone latest mapbox-gl-native (58d773472274e06cab65a55842ce9ba5a6a7e140)
 2. `make offline`

### Expected behavior

Build is successful

### Actual behavior

```
Ld /Users/username/Documents/Mapping/mapbox-gl-native/build/macos/Release/mbgl-offline normal x86_64 (in target: mbgl-offline)
    cd /Users/username/Documents/Mapping/mapbox-gl-native
    export MACOSX_DEPLOYMENT_TARGET=10.11
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -L/Users/username/Documents/Mapping/mapbox-gl-native/build/macos/Release -F/Users/username/Documents/Mapping/mapbox-gl-native/build/macos/Release -filelist /Users/username/Documents/Mapping/mapbox-gl-native/build/macos/mbgl.build/Release/mbgl-offline.build/Objects-normal/x86_64/mbgl-offline.LinkFileList -mmacosx-version-min=10.11 -Wl,-search_paths_first -Wl,-headerpad_max_install_names /Users/username/Documents/Mapping/mapbox-gl-native/build/macos/Release/libmbgl-core.a /Users/username/Documents/Mapping/mapbox-gl-native/build/macos/Release/libmbgl-filesource.a /Users/username/Documents/Mapping/mapbox-gl-native/build/macos/Release/libmbgl-loop-darwin.a /Users/username/Documents/Mapping/mapbox-gl-native/build/macos/Release/libmbgl-core.a /Users/username/Documents/Mapping/mapbox-gl-native/build/macos/Release/libicu.a -framework OpenGL -lz -framework CoreText -framework CoreGraphics -framework ImageIO -framework CoreServices -framework SystemConfiguration -lsqlite3 -framework Foundation -Xlinker -dependency_info -Xlinker /Users/username/Documents/Mapping/mapbox-gl-native/build/macos/mbgl.build/Release/mbgl-offline.build/Objects-normal/x86_64/mbgl-offline_dependency_info.dat -o /Users/username/Documents/Mapping/mapbox-gl-native/build/macos/Release/mbgl-offline
undef: __ZN4mbgl12LayerManager3getEv
Undefined symbols for architecture x86_64:
  "mbgl::LayerManager::get()", referenced from:
      mbgl::style::Parser::parseLayer(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const&, std::__1::unique_ptr<mbgl::style::Layer, std::__1::default_delete<mbgl::style::Layer> >&) in lto.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** BUILD FAILED **
```

### Configuration

**Mapbox SDK versions:** latest head (58d773472274e06cab65a55842ce9ba5a6a7e140)
**iOS/macOS versions:** 10.14.6
**Device/simulator models:** n/a
**Xcode version:** 10.2.1

I can fix this error with:
```diff
diff --git a/platform/macos/config.cmake b/platform/macos/config.cmake
index 0394d3880..378657bfd 100644
--- a/platform/macos/config.cmake
+++ b/platform/macos/config.cmake
@@ -17,6 +17,10 @@ macro(mbgl_platform_core)
         )
     endif()
 
+    target_sources(mbgl-core
+        PRIVATE platform/default/src/mbgl/layermanager/layer_manager.cpp
+    )
+
     target_compile_options(mbgl-core
         PRIVATE -fobjc-arc
     )
```
I'm happy to put a pull request in but I doubt this is the best way of fixing the problem, since I think there is a darwin specific version of layermanager. The mbgl-offline tool seems to work fine with the generic version though.

--- a/platform/macos/config.cmake
+++ b/platform/macos/config.cmake
@@ -70,6 +70,10 @@ endmacro()
 
 
 macro(mbgl_platform_offline)
+    target_sources(mbgl-offline
+        PRIVATE platform/default/src/mbgl/layermanager/layer_manager.cpp
+    )
+
     target_link_libraries(mbgl-offline
         PRIVATE mbgl-filesource
         PRIVATE mbgl-loop-darwin

